### PR TITLE
More Setup

### PR DESCRIPTION
**Pull request by Apidcloud**
This PR will be focused on upgrading the current structure to use typescript instead.
It should also make lerna work with semantic-release (#39)

**Notes:**

- Use typescript for type checking and generation of _.d.ts_ files
- Continue using _Babel_ as the compiler, but for typescript
- Continue using _Eslint_ for linting (instead of going to _ts-lint_)
- Continue using _Prettier_ for formatting
- Continue using _Jest_ for testing, but for typescript
- Continue using _Webpack_ for bundling

Since we are also using _lerna_, we have two options, as per their [documentation](https://github.com/lerna/lerna/blob/master/README.md#how-it-works):

- independent versioning, where each package has its own version
- fixed/lock versioning, where every package gets the same version (repos such as Babel use that.)

I think the independent versioning might make more sense (read the docs linked above), but we need to configure some more stuff with the CI: https://github.com/atlassian/lerna-semantic-release

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -5,14 +5,14 @@ module.exports = api => {
   let plugins = [];
 
   switch (env) {
-    case "test": {
-      presets = ["@babel/preset-env"];
+    case 'test': {
```
Comment (joafalves) [inline babel.config.js:8]: Use "" instead

File: config/webpack.config.js [hunk 2]
```diff
@@ -1,34 +1,37 @@
-const path = require("path");
-const glob = require("glob");
+const path = require('path');
+const glob = require('glob');
+const ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
 
 module.exports = {
-  mode: "development",
+  mode: 'development',
```
Comment (joafalves) [inline config/webpack.config.js]: Same as above..

File: lerna.json [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {
   "packages": ["packages/*"],
-  "version": "0.0.1"
+  "version": "0.0.1",
```
Comment (joafalves) [inline lerna.json:3]: Version should be based on release branches and tags. It should not be placed on code. Check git versioning

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "files": [],
+  "references": [
+    {
+      "path": "packages/commons"
```
Comment (joafalves) [inline tsconfig.json:5]: "module/" is more appropriate, these ARE modules.

File: babel.config.js (hunk 1 above)
Comment (Apidcloud) [reply babel.config.js:8]: Oh, I should just enable prettier on these files I guess.

File: lerna.json (hunk 3 above)
Comment (Apidcloud) [reply lerna.json:3]: The versioning goes in the code. But lerna cli handles the bumping based on the commits. It updates the files automatically. But it's something we need to try some time, after setting up the CI with some environment variables, etc. It's explained in the lerna commitizen repo.

File: tsconfig.json (hunk 4 above)
Comment (Apidcloud) [reply tsconfig.json:5]: Hmm the convention in these monorepos is just to call them packages, which they still are. Babel uses the same system even though they are all plugins or presets. I think it makes more sense to call them something they all are: packages.

**Pull request merged**